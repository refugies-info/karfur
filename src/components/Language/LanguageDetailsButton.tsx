import * as React from "react";
import styled from "styled-components/native";
import { StyledTextSmallBold, StyledTextSmall } from "../StyledText";
import { theme } from "../../theme";
import { Flag } from "./Flag";
import { RowContainer } from "../BasicComponents";

const MainContainer = styled.TouchableOpacity`
  background: ${(props: { isSelected: any }) =>
    props.isSelected ? theme.colors.black : theme.colors.white};
  border-radius: ${theme.radius * 2}px;
  padding: ${theme.margin * 2}px;
  margin-vertical: ${theme.margin}px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  flex-wrap: wrap;
  elevation: 1;
  box-shadow: 0px 8px 16px rgba(33, 33, 33, 0.24);
`;

const StyledTextBold = styled(StyledTextSmallBold)`
  text-align: left;
  margin-left: ${theme.margin}px;
  color: ${(props: { isSelected: boolean }) =>
    props.isSelected ? theme.colors.white : theme.colors.black};
`;

const StyledText = styled(StyledTextSmall)`
  text-align: left;
  color: ${(props: { isSelected: boolean }) =>
    props.isSelected ? theme.colors.white : theme.colors.black};
`;

const FlagBackground = styled.View`
  margin: 4px;
  background-color: ${theme.colors.white};
  width: 22px;
  height: 17px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 4px;
`;

interface Props {
  langueFr: string;
  langueLoc: string;
  onPress: () => void;
  isSelected?: boolean;
}
export const LanguageDetailsButton = (props: Props) => (
  <MainContainer
    onPress={props.onPress}
    testID={"test-language-button-" + props.langueFr}
    isSelected={props.isSelected}
  >
    <RowContainer>
      <FlagBackground>
        <Flag langueFr={props.langueFr} />
      </FlagBackground>
      <StyledTextBold isSelected={props.isSelected}>
        {props.langueFr}
      </StyledTextBold>
      {props.langueFr !== "Fran√ßais" && (
        <RowContainer>
          <StyledText isSelected={props.isSelected}>{" - "}</StyledText>
          <StyledText isSelected={props.isSelected}>
            {props.langueLoc}
          </StyledText>
        </RowContainer>
      )}
    </RowContainer>
  </MainContainer>
);
