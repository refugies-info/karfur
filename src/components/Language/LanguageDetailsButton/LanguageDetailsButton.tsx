import React from "react";
import styled from "styled-components/native";
import { StyledTextSmallBold, StyledTextSmall } from "../../StyledText";
import { Flag } from "../Flag";
import { useTranslationWithRTL } from "../../../hooks/useTranslationWithRTL";
import { ChoiceButton, ChoiceButtonProps } from "../../formulaire";
import { Columns } from "../../layout";

const StyledTextBold = styled(StyledTextSmallBold)`
  text-align: left;
  margin-left: ${({ theme }) => (!theme.i18n.isRTL ? theme.margin * 2 : 0)}px;
  margin-right: ${({ theme }) => (theme.i18n.isRTL ? theme.margin * 2 : 0)}px;
  color: ${({ theme }) => theme.colors.black};
`;

const StyledText = styled(StyledTextSmall)`
  text-align: left;
  color: ${({ theme }) => theme.colors.darkGrey};
`;

const FlagBackground = styled.View`
  margin: 4px;
  background-color: ${({ theme }) => theme.colors.white};
  width: 22px;
  height: 17px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 4px;
  ${({ theme }) => theme.shadows.sm}
`;

export interface LanguageDetailsButtonProps {
  flatStyle?: boolean;
  hideRadio?: boolean;
  isSelected?: boolean;
  langueFr: string;
  langueLoc: string;
  onPress: ChoiceButtonProps["onPress"];
}

const LanguageDetailsButton = ({
  flatStyle,
  hideRadio,
  isSelected,
  langueFr,
  langueLoc,
  onPress,
}: LanguageDetailsButtonProps) => {
  const { isRTL } = useTranslationWithRTL();

  return (
    <ChoiceButton
      accessibilityRole="button"
      flatStyle={flatStyle}
      hideRadio={hideRadio}
      isSelected={!!isSelected}
      onPress={onPress}
      testID={"test-language-button-" + langueFr}
    >
      <Columns RTLBehaviour layout="auto">
        <FlagBackground>
          <Flag langueFr={langueFr} />
        </FlagBackground>
        <StyledTextBold>{langueLoc}</StyledTextBold>
        {langueFr !== "Fran√ßais" && (
          <StyledText>
            {!isRTL ? " - " + langueFr : langueFr + " - "}
          </StyledText>
        )}
      </Columns>
    </ChoiceButton>
  );
};

export default LanguageDetailsButton;
