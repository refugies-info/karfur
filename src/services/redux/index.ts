export * from "./SelectedContent";
export * from "./Themes";
export * from "./User";
