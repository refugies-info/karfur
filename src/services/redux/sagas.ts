import { SagaIterator } from "redux-saga";
import { fork } from "redux-saga/effects";
import usersSaga from "./User/user.saga";
import languagesSaga from "./Languages/languages.saga";
import contentsSaga from "./Contents/contents.saga";
import selectedContentSaga from "./SelectedContent/selectedContent.saga";

export function* rootSaga(): SagaIterator {
  yield fork(usersSaga);
  yield fork(languagesSaga);
  yield fork(contentsSaga);
  yield fork(selectedContentSaga);
}
