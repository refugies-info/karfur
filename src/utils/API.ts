import axios from "react-native-axios";
import { getEnvironment } from "../libs/getEnvironment";

const dbURL = getEnvironment().dbUrl;
const siteSecret = getEnvironment().siteSecret;

const apiCaller = axios.create({
  baseURL: dbURL,
  headers: {
    "Content-Type": "application/json",
    "site-secret": siteSecret,
  },
});

export const getLanguages = () => apiCaller.get("/langues/getLanguages");

export const getCitiesFromGoogleAPI = (text: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/autocomplete/json?input=${text}&language=fr&types=(cities)&components=country:fr&key=${process.env.GOOGLE_API_KEY}`
  );

export const getCityDetailsFromGoogleAPI = (placeId: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/details/json?place_id=${placeId}&fields=address_component&key=${process.env.GOOGLE_API_KEY}`
  );

export const getPlaceIdFromLocationFromGoogleAPI = (
  longitude: number,
  latitude: number
) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latitude},${longitude}&key=${process.env.GOOGLE_API_KEY}`
  );

export const getContentsForApp = ({
  locale,
  age,
  department,
  frenchLevel,
}: {
  locale: string;
  age: string | null;
  department: string | null;
  frenchLevel: string | null;
}) => {
  const route =
    `/dispositifs/getContentsForApp?locale=${locale}` +
    (age ? `&age=${age}` : "") +
    (department ? `&department=${department}` : "") +
    (frenchLevel ? `&frenchLevel=${frenchLevel}` : "");

  return apiCaller.get(route);
};

export const get_dispositif = (params = {}) =>
  apiCaller.post("/dispositifs/get_dispositif", params);
