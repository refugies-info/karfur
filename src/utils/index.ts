export { default as withProps } from "./withProps";
