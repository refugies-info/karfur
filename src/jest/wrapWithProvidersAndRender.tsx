import React from "react";
import { render } from "react-native-testing-library";
import { Provider } from "react-redux";
import { createStore } from "redux";
import {
  initialRootStateFactory,
  rootReducer,
  RootState,
} from "../services/redux/reducers";

interface WrapWithProvidersAndRenderParams {
  Component: React.FunctionComponent<unknown>;
  compProps?: Record<string, unknown>;
  reduxState?: Partial<RootState>;
}

/**
 * Provide all the ugly-to-set-up providers for your component to be ready to test
 * @param Component
 * @param compProps properties passed on to [getPropsWithNavigation()] to gen navigation props
 * @param reduxState defaults to initialRootState
 */
export function wrapWithProvidersAndRender({
  Component,
  compProps = {},
  reduxState = initialRootStateFactory(),
}: WrapWithProvidersAndRenderParams) {
  const store = createStore(rootReducer, reduxState);

  const componentWithReduxAndI18n = (
    <Provider store={store}>
      <Component {...compProps} />
    </Provider>
  );

  return render(componentWithReduxAndI18n);
}
