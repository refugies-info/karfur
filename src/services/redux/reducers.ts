import { Action, combineReducers } from "redux";
import {
  loadingStatusReducer,
  LoadingStatusState,
  initialLoadingState,
} from "./LoadingStatus/loadingStatus.reducer";
import {
  languagesReducer,
  LanguageState,
  initialLanguageState,
} from "./Languages/languages.reducer";
import { userReducer, UserState, initialUserState } from "./User/user.reducer";
import {
  contentsReducer,
  ContentsState,
  initialContentsState,
} from "./Contents/contents.reducer";
import {
  SelectedContentState,
  initialSelectedContentState,
  selectedContentReducer,
} from "./SelectedContent/selectedContent.reducer";
import {
  NeedState,
  initialNeedState,
  needsReducer,
} from "./Needs/needs.reducer";
import {
  groupedContentsReducer,
  GroupedContentsState,
  initialGroupedContentsState,
} from "./ContentsGroupedByNeeds/contentsGroupedByNeeds.reducer";
import {
  VoiceOverState,
  initialVoiceOverState,
  voiceOverReducer,
} from "./VoiceOver/voiceOver.reducer";
const appReducer = combineReducers({
  loadingStatus: loadingStatusReducer,
  languages: languagesReducer,
  user: userReducer,
  contents: contentsReducer,
  selectedContent: selectedContentReducer,
  needs: needsReducer,
  groupedContents: groupedContentsReducer,
  voiceOver: voiceOverReducer,
});

export interface RootState {
  loadingStatus: LoadingStatusState;
  languages: LanguageState;
  user: UserState;
  contents: ContentsState;
  selectedContent: SelectedContentState;
  needs: NeedState;
  groupedContents: GroupedContentsState;
  voiceOver: VoiceOverState;
}

type RootReducer = ReturnType<typeof appReducer>;

export const initialRootStateFactory = (): RootState => ({
  loadingStatus: initialLoadingState,
  languages: initialLanguageState,
  user: initialUserState,
  contents: initialContentsState,
  selectedContent: initialSelectedContentState,
  needs: initialNeedState,
  groupedContents: initialGroupedContentsState,
  voiceOver: initialVoiceOverState,
});

export const rootReducer = (
  state: RootState | undefined,
  action: Action<any>
): RootReducer =>
  // @ts-ignore
  appReducer(state, action);
