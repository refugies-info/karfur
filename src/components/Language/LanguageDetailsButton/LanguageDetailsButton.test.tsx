import React from "react";
import { act, fireEvent } from "react-native-testing-library";
import LanguageDetailsButton from "./LanguageDetailsButton";
import { render } from "../../utils/tests";
import { wrapWithProvidersAndRender } from "../../../jest/wrapWithProvidersAndRender";
import { initialRootStateFactory } from "../../../services/redux/reducers";

describe("LanguageDetailsButton snapshot test suite", () => {
  it("should render without bug", async () => {
    const test = await render(
      <LanguageDetailsButton
        isSelected
        langueFr="Français"
        langueLoc="Français"
        onPress={() => null}
      />
    );
    expect(test).toMatchSnapshot();
  });

  it("should render not selected", async () => {
    const test = await render(
      <LanguageDetailsButton
        isSelected={false}
        langueFr="Français"
        langueLoc="Français"
        onPress={() => null}
      />
    );
    expect(test).toMatchSnapshot();
  });

  it("should render with local language", async () => {
    const test = await render(
      <LanguageDetailsButton
        isSelected
        langueFr="Anglais"
        langueLoc="English"
        onPress={() => null}
      />
    );
    expect(test).toMatchSnapshot();
  });

  it("should render with radio hided", async () => {
    const test = await render(
      <LanguageDetailsButton
        isSelected
        langueFr="Anglais"
        langueLoc="English"
        onPress={() => null}
        hideRadio
      />
    );
    expect(test).toMatchSnapshot();
  });

  it("should react on press", async () => {
    const mockOnPress = jest.fn();

    const component = wrapWithProvidersAndRender({
      Component: LanguageDetailsButton,
      reduxState: {
        ...initialRootStateFactory(),
      },
      compProps: {
        isSelected: true,
        langueFr: "Anglais",
        langueLoc: "English",
        onPress: mockOnPress,
        hideRadio: true,
      },
    });

    const Button = component.getByTestId("test-language-button-Anglais");
    act(() => {
      fireEvent.press(Button);
    });

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
