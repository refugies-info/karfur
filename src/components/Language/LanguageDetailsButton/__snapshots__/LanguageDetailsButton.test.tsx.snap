// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageDetailsButton snapshot test suite should render not selected 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#FFFFFF",
      "borderRadius": 12,
      "borderWidth": 2,
      "display": "flex",
      "elevation": 6,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-between",
      "marginBottom": 24,
      "opacity": 1,
      "paddingBottom": 14,
      "paddingLeft": 14,
      "paddingRight": 14,
      "paddingTop": 14,
      "shadowColor": "#212121",
      "shadowOffset": {
        "height": 1,
        "width": 1,
      },
      "shadowOpacity": 0.16,
      "shadowRadius": 8,
    }
  }
  testID="test-language-button-Français"
>
  <View
    RTLBehaviour={true}
    collapsable={false}
    horizontalAlign="flex-start"
    style={
      {
        "alignItems": "center",
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "flex-start",
      }
    }
    verticalAlign="center"
  >
    <View
      style={
        [
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "marginRight": 8,
          },
        ]
      }
    >
      <View
        RTLBehaviour={true}
        collapsable={false}
        horizontalAlign="flex-start"
        style={
          {
            "alignItems": "center",
            "flexBasis": "auto",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "flex-start",
          }
        }
        verticalAlign="center"
      >
        <View
          style={
            [
              {
                "flexBasis": "auto",
                "flexGrow": 0,
                "marginRight": 8,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "display": "flex",
                  "elevation": 2,
                  "height": 17,
                  "justifyContent": "center",
                  "marginBottom": 4,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 4,
                  "shadowColor": "#212121",
                  "shadowOffset": {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.4,
                  "shadowRadius": 2,
                  "width": 22,
                },
              ]
            }
          >
            <react-native-svg
              height={15}
              width={20}
            />
          </View>
        </View>
        <View
          style={
            [
              {
                "flexBasis": "auto",
                "flexGrow": 0,
                "marginRight": 8,
              },
            ]
          }
        >
          <Text
            isRTL={false}
            selectable={true}
            style={
              [
                {
                  "color": "#212121",
                  "fontFamily": "circularBold",
                  "fontSize": 16,
                  "lineHeight": 20,
                  "marginLeft": 16,
                  "marginRight": 0,
                  "textAlign": "left",
                },
              ]
            }
          >
            Français
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexBasis": "auto",
            "flexGrow": 0,
          },
        ]
      }
    >
      <View
        selected={false}
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#5E5E5E",
              "borderRadius": 12,
              "borderWidth": 1,
              "height": 24,
              "justifyContent": "center",
              "width": 24,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`LanguageDetailsButton snapshot test suite should render with local language 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#DAE0FB",
      "borderColor": "#0421B1",
      "borderRadius": 12,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-between",
      "marginBottom": 24,
      "opacity": 1,
      "paddingBottom": 14,
      "paddingLeft": 14,
      "paddingRight": 14,
      "paddingTop": 14,
    }
  }
  testID="test-language-button-Anglais"
>
  <View
    RTLBehaviour={true}
    collapsable={false}
    horizontalAlign="flex-start"
    style={
      {
        "alignItems": "center",
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "flex-start",
      }
    }
    verticalAlign="center"
  >
    <View
      style={
        [
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "marginRight": 8,
          },
        ]
      }
    >
      <View
        RTLBehaviour={true}
        collapsable={false}
        horizontalAlign="flex-start"
        style={
          {
            "alignItems": "center",
            "flexBasis": "auto",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "flex-start",
          }
        }
        verticalAlign="center"
      >
        <View
          style={
            [
              {
                "flexBasis": "auto",
                "flexGrow": 0,
                "marginRight": 8,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "display": "flex",
                  "elevation": 2,
                  "height": 17,
                  "justifyContent": "center",
                  "marginBottom": 4,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 4,
                  "shadowColor": "#212121",
                  "shadowOffset": {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.4,
                  "shadowRadius": 2,
                  "width": 22,
                },
              ]
            }
          >
            <react-native-svg
              height={15}
              width={20}
            />
          </View>
        </View>
        <View
          style={
            [
              {
                "flexBasis": "auto",
                "flexGrow": 0,
                "marginRight": 8,
              },
            ]
          }
        >
          <Text
            isRTL={false}
            selectable={true}
            style={
              [
                {
                  "color": "#212121",
                  "fontFamily": "circularBold",
                  "fontSize": 16,
                  "lineHeight": 20,
                  "marginLeft": 16,
                  "marginRight": 0,
                  "textAlign": "left",
                },
              ]
            }
          >
            English
          </Text>
        </View>
        <View
          style={
            [
              {
                "flexBasis": "auto",
                "flexGrow": 0,
              },
            ]
          }
        >
          <Text
            isRTL={false}
            selectable={true}
            style={
              [
                {
                  "color": "#5E5E5E",
                  "fontFamily": "circularBook",
                  "fontSize": 16,
                  "lineHeight": 20,
                  "textAlign": "left",
                },
              ]
            }
          >
             - Anglais
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexBasis": "auto",
            "flexGrow": 0,
          },
        ]
      }
    >
      <View
        selected={true}
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#0421B1",
              "borderColor": "transparent",
              "borderRadius": 12,
              "borderWidth": 1,
              "height": 24,
              "justifyContent": "center",
              "width": 24,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="16"
            bbWidth="16"
            fill="#FFFFFF"
            focusable={false}
            height={16}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={16}
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGRect
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    height="24"
                    opacity={0}
                    width="24"
                    x="0"
                    y="0"
                  />
                  <RNSVGPath
                    d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`LanguageDetailsButton snapshot test suite should render with radio hided 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#DAE0FB",
      "borderColor": "#0421B1",
      "borderRadius": 12,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-between",
      "marginBottom": 24,
      "opacity": 1,
      "paddingBottom": 14,
      "paddingLeft": 14,
      "paddingRight": 14,
      "paddingTop": 14,
    }
  }
  testID="test-language-button-Anglais"
>
  <View
    RTLBehaviour={true}
    collapsable={false}
    horizontalAlign="flex-start"
    style={
      {
        "alignItems": "center",
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "flex-start",
      }
    }
    verticalAlign="center"
  >
    <View
      style={
        [
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "marginRight": 8,
          },
        ]
      }
    >
      <View
        RTLBehaviour={true}
        collapsable={false}
        horizontalAlign="flex-start"
        style={
          {
            "alignItems": "center",
            "flexBasis": "auto",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "flex-start",
          }
        }
        verticalAlign="center"
      >
        <View
          style={
            [
              {
                "flexBasis": "auto",
                "flexGrow": 0,
                "marginRight": 8,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "display": "flex",
                  "elevation": 2,
                  "height": 17,
                  "justifyContent": "center",
                  "marginBottom": 4,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 4,
                  "shadowColor": "#212121",
                  "shadowOffset": {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.4,
                  "shadowRadius": 2,
                  "width": 22,
                },
              ]
            }
          >
            <react-native-svg
              height={15}
              width={20}
            />
          </View>
        </View>
        <View
          style={
            [
              {
                "flexBasis": "auto",
                "flexGrow": 0,
                "marginRight": 8,
              },
            ]
          }
        >
          <Text
            isRTL={false}
            selectable={true}
            style={
              [
                {
                  "color": "#212121",
                  "fontFamily": "circularBold",
                  "fontSize": 16,
                  "lineHeight": 20,
                  "marginLeft": 16,
                  "marginRight": 0,
                  "textAlign": "left",
                },
              ]
            }
          >
            English
          </Text>
        </View>
        <View
          style={
            [
              {
                "flexBasis": "auto",
                "flexGrow": 0,
              },
            ]
          }
        >
          <Text
            isRTL={false}
            selectable={true}
            style={
              [
                {
                  "color": "#5E5E5E",
                  "fontFamily": "circularBook",
                  "fontSize": 16,
                  "lineHeight": 20,
                  "textAlign": "left",
                },
              ]
            }
          >
             - Anglais
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`LanguageDetailsButton snapshot test suite should render without bug 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#DAE0FB",
      "borderColor": "#0421B1",
      "borderRadius": 12,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-between",
      "marginBottom": 24,
      "opacity": 1,
      "paddingBottom": 14,
      "paddingLeft": 14,
      "paddingRight": 14,
      "paddingTop": 14,
    }
  }
  testID="test-language-button-Français"
>
  <View
    RTLBehaviour={true}
    collapsable={false}
    horizontalAlign="flex-start"
    style={
      {
        "alignItems": "center",
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "flex-start",
      }
    }
    verticalAlign="center"
  >
    <View
      style={
        [
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "marginRight": 8,
          },
        ]
      }
    >
      <View
        RTLBehaviour={true}
        collapsable={false}
        horizontalAlign="flex-start"
        style={
          {
            "alignItems": "center",
            "flexBasis": "auto",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "flex-start",
          }
        }
        verticalAlign="center"
      >
        <View
          style={
            [
              {
                "flexBasis": "auto",
                "flexGrow": 0,
                "marginRight": 8,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "display": "flex",
                  "elevation": 2,
                  "height": 17,
                  "justifyContent": "center",
                  "marginBottom": 4,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 4,
                  "shadowColor": "#212121",
                  "shadowOffset": {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.4,
                  "shadowRadius": 2,
                  "width": 22,
                },
              ]
            }
          >
            <react-native-svg
              height={15}
              width={20}
            />
          </View>
        </View>
        <View
          style={
            [
              {
                "flexBasis": "auto",
                "flexGrow": 0,
                "marginRight": 8,
              },
            ]
          }
        >
          <Text
            isRTL={false}
            selectable={true}
            style={
              [
                {
                  "color": "#212121",
                  "fontFamily": "circularBold",
                  "fontSize": 16,
                  "lineHeight": 20,
                  "marginLeft": 16,
                  "marginRight": 0,
                  "textAlign": "left",
                },
              ]
            }
          >
            Français
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexBasis": "auto",
            "flexGrow": 0,
          },
        ]
      }
    >
      <View
        selected={true}
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#0421B1",
              "borderColor": "transparent",
              "borderRadius": 12,
              "borderWidth": 1,
              "height": 24,
              "justifyContent": "center",
              "width": 24,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="16"
            bbWidth="16"
            fill="#FFFFFF"
            focusable={false}
            height={16}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={16}
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGRect
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    height="24"
                    opacity={0}
                    width="24"
                    x="0"
                    y="0"
                  />
                  <RNSVGPath
                    d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</View>
`;
